models/view/summary.go

```diff
@@ -48,6 +60,30 @@ func NewDailyProjectStats(summaries []*models.Summary) []*DailyProjectsViewModel
 	return dailyProjects
 }
 
+func NewTimelineViewModel(durations models.Durations) []*TimelineViewModel {
+	timeline := make([]*TimelineViewModel, 0)
+	// Group by project
```

Comment:
Using [`GroupWith`](https://github.com/duke-git/lancet/blob/main/docs/en/api/packages/slice.md#GroupWith) for that would probably a bit more elegant and also more efficient than two nested loops, what do you think?